src/main/java/org/embulk/output/mailchimp/MailChimpRecordBuffer.java

```diff
@@ -129,7 +130,11 @@ public TaskReport commitWithTaskReportUpdated(TaskReport taskReport)
                 uniqueRecords = new ArrayList<>();
                 duplicatedRecords = new ArrayList<>();
             }
-
+            // When atomic upsert is true, client expects all records are done properly.
+            if (task.getAtomicUpsert() && errorCount > 0) {
+                LOG.info("Job requires atomic operation for all records. And there were {} errors in processing => Error as job's status", errorCount);
+                throw Throwables.propagate(new DataException("Some records are not properly processed at MailChimp. See log for detail"));
```

Comment:
If we throw DataException here, it will prevent from other task reports to be completed. 

We should consider the following updates :
Exec.newTaskReport().set("pushed", totalCount).set("error_count", error_count);

Then, in `MailChimpOutputPluginDelegate.egestEmbulkData` we count total error to throw DataException 

>      long totalInserted = 0;
        long totalErrors = 0;
        for (TaskReport taskReport : taskReports) {
            if (taskReport.has("pushed")) {
                totalInserted += taskReport.get(Long.class, "pushed");
            }
            if (taskReport.has("error_count")) {
                 totalErrors += taskReport.get(Long.class, "error_count");
            }
        }
        LOG.info("Pushed completed. {} records", totalInserted);
       if (task.getAtomicUpsert() && totalErrors > 0) {
            LOG.info("Job requires atomic operation for all records. And there were {} errors in processing => Error as job's status", totalErrors);
                 throw new DataException("Some records are not properly processed at MailChimp. See log for detail");
      }